### PR TITLE
boot_services.rs: Add a test for allocate pool

### PR DESCRIPTION
**Pull request by dachihy**
## Description

Adds a test for allocate pool wrapper.

- [ ] Impacts functionality?
- [ ] Impacts security?
- [ ] Breaking change?
- [x] Includes tests?
- [ ] Includes documentation?

## How This Was Tested

Run Test with PASS result

## Integration Instructions

N/A

### CODE REVIEW

**Review by joschock — COMMENTED** [review]
the `_` usually is used to mark an unused parameter - but in this case, you are setting it to a value for the exit, so you can just name it `buffer` instead of `_buffer`.

File: boot_services/src/boot_services.rs [hunk 1]
```diff
@@ -1487,4 +1487,21 @@ mod test {
...
+        extern "efiapi" fn efi_allocate_pool(mem_type: efi::MemoryType, size: usize, mut _buffer: *mut *mut c_void) -> efi::Status {
+            let expected_mem_type: efi::MemoryType = MemoryType::MEMORY_MAPPED_IO.into();
+            assert_eq!(mem_type, expected_mem_type);
+            assert_eq!(size, 10);
+            _buffer = 0x55AA as *mut *mut c_void;
+            efi::Status::SUCCESS
+        }
+
+        let status = boot_services.allocate_pool(MemoryType::MEMORY_MAPPED_IO, 10);
+        let _expected_buffer = 0x55AA as *mut u8;
```
Comment (joschock) [inline boot_services/src/boot_services.rs:1504]: Same with `_expected_buffer` - you can name it `expected_buffer` - preceding `_` means "unused" but that isn't the case here.

**Review by joschock — APPROVED** [review]

**Review by makubacki — APPROVED** [review]

**Comment by dachihy** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by makubacki** [comment]
Please keep the PR template checkboxes in the PR description in the future. I've added them back and checked the "test" box, so the label gets added for that.

**Pull request merged**